src/webapp/components/landing-page-edit-dialog/useImageFileUpload.tsx

```diff
@@ -0,0 +1,93 @@
+import { ChangeEvent, useCallback, useState } from "react";
+import { useAppContext } from "../../contexts/app-context";
+import { LandingNode } from "../../../domain/entities/LandingNode";
+import i18n from "../../../locales";
+
+export default function useImageFileUpload(updateNode: (value: React.SetStateAction<LandingNode>) => void) {
+    const { compositionRoot } = useAppContext();
+
+    const [faviconWarnings, setFaviconWarnings] = useState<string[]>([]);
+
+    const processImageFile = (file: File, fileType: keyof LandingNode) => {
+        const reader = new FileReader();
+        reader.onload = e => {
+            const img = new Image();
+            img.onload = () => {
+                const width = img.width;
+                const height = img.height;
+                const aspectRatio = width / height;
+
+                setFaviconWarnings([]);
+                if (fileType === "favicon") {
+                    const aspectRatioWarnings = [
```

Comment:
Good work! **[No need to change]** There is a lot of ways we could declare this, another way a bit more readable would be for example (**that I didn't test it**, it should work but I didn't check):

```typescript
const warnings = _.compact([
    !isAspectRatioWithinMargin(aspectRatio) &&
        i18n.t("Please ensure that your favicon has a {{aspectRatio}} aspect ratio.", {
            aspectRatio: FAVICON_ASPECT_RATIO.fraction,
        }),
    !(isWithinMargin(width) && isWithinMargin(height)) &&
        i18n.t("Please use an icon of {{max}}x{{max}} pixels or smaller.", {
            max: FAVICON_MAX_SIZE,
        }),
]);

setFaviconWarnings(warnings);
```